connector/connect/common/src/main/protobuf/spark/connect/relations.proto

```diff
@@ -304,6 +305,24 @@ message LocalRelation {
   // Local collection data serialized into Arrow IPC streaming format which contains
   // the schema of the data.
   bytes data = 1;
+
+  // (Optional) The user provided schema.
+  //
+  // The Sever side will update the column names and data types according to this schema.
+  oneof schema {
+
+    DataType datatype = 2;
+
+    // Server will use Catalyst parser to parse this string to DataType.
+    string datatype_str = 3;
+
+    // Column names
+    StringList cols = 4;
```

Comment:
I am a bit confused on this one: what would be the types for such column names? Are the types inferred but the names are provided? Can the behavior be documented here?

Reply:
they are used for renaming, I'm think removing it and directly use `toDF` in Client